lib/test_queue/iterator.rb

```diff
@@ -13,13 +13,22 @@ def initialize(sock, suites, filter=nil)
         @tcp_address = $1
         @tcp_port = $2.to_i
       end
+      @failures = 0
     end
 
     def each
       fail "already used this iterator. previous caller: #@done" if @done
 
       while true
-        client = connect_to_master('POP')
+        # If we've hit too many failures in one worker, assume the entire
+        # test suite is broken, and notify master so the run
+        # can be immediately halted.
+        if ENV["TEST_QUEUE_EARLY_FAILURE_LIMIT"] && @failures >= ENV["TEST_QUEUE_EARLY_FAILURE_LIMIT"].to_i
```

Comment:
Good idea; switched to using `Integer()`, and moved the parsing into `Runner`'s initialization, with an informative message if parsing fails.